FO: correct submit of carrier form

Shoprunners opened the pull request:
submitting the form via $(this) instead of $(#form)submits the button which results the form. The difference is now, that the processCarrier methods get executed because they rely on the name of the button (processCarrier...). This was not executed before.

Commit:
FO: correct submit of carrier form

submitting the form via $(this) instead of $(#form)submits the button which results the form. The difference is now, that the processCarrier methods get executed because they rely on the name of the button (processCarrier...). This was not executed before.

aleeks reviewed (dismissed):
It makes sense to me, need to be tested.

Commit:
FO: call carrier hooks with advancanced eu enabled

in certain circumstance this bugfix (this).submit() did not work so we had to add an hidden input field to improve it

PierreRambaud commented on views/js/fo_aeuc_tnc.js:
```diff
@@ -49,7 +49,14 @@ $(document).ready(function(){
         }
-        $("#form").submit();
+        $('<input>').attr({
+			type: 'hidden',
```
Don't mix tab and spaces please, weird indent.

PierreRambaud reviewed (changes requested):
See comment

Progi1984 commented:
Ping @Shoprunners

Shoprunners commented:
I corrected the spaces and tabs, hopefully to the right branch?

Progi1984 commented:
@Shoprunners I don't see the change actually. So it doesn't seem on the right branch.

Shoprunners committed:
FO: correct submit of carrier form 

Corrected line indents and year in licence

Shoprunners commented:
> @Shoprunners I don't see the change actually. So it doesn't seem on the right branch.

I hope, i got the right one this time, sorry, not yet very used to the review and resubmit process

Progi1984 commented:
@Shoprunners No problem ;)

sarahdib commented:
Hello @Shoprunners 
Could you please provide us a test case scenario in order to test this pr ?

Thank you :)

Shoprunners commented:
Hi,
well, this bugfix is for i.e. this:

https://www.prestashop.com/forums/topic/626896-aeuc-module-and-actioncarrierprocess/

Hope this helps.

sarahdib commented:
Hello @Shoprunners 

We need a scenario not technical in order to test this pr.
Like : 
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

Thank you

Shoprunners commented:
Hi, 
you need a  module which hooks into official hook "actionCarrierProcess". When ps_legalcomplicance is installed, this hook will never execute.

It is going to be executed with this bugfix.

Should be a bugfix for this: https://github.com/PrestaShop/PrestaShop/issues/12780 as well.

sarahdib commented:
Hello @Shoprunners 
Could you provide us the module in order to test?

Thank you

sarahdib commented:
@Shoprunners 

We try to check the pr with a module and nothing change before and after the PR.

Could you give us a precise scenario in order to test ?

Thank you

Shoprunners commented:
Hi, I think, you can close this. Because the code in the core changed since 2017 and it is not reproducable in the latest versions.
